src/App.tsx

```diff
@@ -1,24 +1,110 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
-
-const USER_ID = 0;
+import React, { useContext, useMemo, useState } from 'react';
+import classNames from 'classnames';
+import { Filter } from './types/Filter';
+import { Todo } from './types/Todo';
+import { TodoList } from './сomponents/TodoList/TodoList';
+import { TodosFilter } from './сomponents/TodoFilter/TodoFilter';
+import { TodoForm } from './сomponents/TodoForm/TodoForm';
+import { TodoItem } from './сomponents/TodoItem/TodoItem';
+import { TodosContext } from './TodosContext';
+import { ErrorMessage } from './types/ErrorMessage';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
-    return <UserWarning />;
-  }
+  const [filter, setFilter] = useState(Filter.ALL);
+
+  const todosContext = useContext(TodosContext);
+
+  const {
+    todos,
+    tempTodo,
+    errorMessage,
+    setError,
+    getFilteredTodos,
+    todoCount,
+    toggleAll,
+    clearCompleted,
+  } = todosContext;
+
+  const filteredTodos: Todo[] = useMemo(() => {
+    return getFilteredTodos(filter);
+  }, [todos, filter]);
 
   return (
-    <section className="section container">
-      <p className="title is-4">
-        Copy all you need from the prev task:
-        <br />
-        <a href="https://github.com/mate-academy/react_todo-app-add-and-delete#react-todo-app-add-and-delete">React Todo App - Add and Delete</a>
-      </p>
-
-      <p className="subtitle">Styles are already copied</p>
-    </section>
+    <div className="todoapp">
+      <h1 className="todoapp__title">todos</h1>
+      <div className="todoapp__content">
+        <header className="todoapp__header">
+
+          {todos.length > 0 && (
```

Comment:
```suggestion
        <header className="todoapp__header">
          {todos.length > 0 && (
```
don't add blank lines between child and parent elements